core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/SearchIndexResourceRequest.java

```diff
@@ -90,7 +74,54 @@ protected final B doExecute(C context) throws IOException {
 		
 		return toCollectionResource(context, hits);
 	}
+
+	private final List<String> configureFieldsToLoad(List<String> fields) {
+		ImmutableSet.Builder<String> additionalFieldsToLoad = ImmutableSet.builder();
+		
+		collectAdditionalFieldsToFetch(additionalFieldsToLoad);
+		
+		// in case of Revisions always include the ID field (if not requested) to avoid low-level error
+		// after configuring the additional field inclusions
+		if (Revision.class.isAssignableFrom(getFrom()) && !fields().contains(Revision.Fields.ID)) {
+			additionalFieldsToLoad.add(Revision.Fields.ID);
+		}
+		
+		final Set<String> additionalFields = additionalFieldsToLoad.build();
+		
+		if (!additionalFields.isEmpty()) {
+			fields = Lists.newArrayList(fields);
+			for (String additionalField : additionalFields) {
+				if (!fields.contains(additionalField)) {
+					fields.add(additionalField);
+				}
+			}
+		}
+		
+		// perform field replacements between known model and index fields, if specified by the subclass
+		final Multimap<String, String> fieldReplacements = collectFieldsToLoadReplacements();
+		if (!fieldReplacements.isEmpty()) {
+			fields = Lists.newArrayList();
+			for (String fieldToLoad : fields) {
+				Collection<String> replacements = fieldReplacements.get(fieldToLoad);
+				if (replacements.isEmpty()) {
+					fields.add(fieldToLoad);
+				} else {
+					fields.addAll(replacements);
+				}
```

Comment:
There is no duplicate check for these fields. A client could request a field that is a replacement of another requested field.

Reply:
Hmm, yes, but that is unnecessary I think, the underlying system can handle multiple requests of the same field as well. Like requesting `id`, `id`, `id` should still be able to load the document without failing or broken. Partial loading (which is only supported in low-level requests at the moment) is another story, but we will figure that one out when we get there. I would leave this like this for now.

Reply:
Then line 85 and the block starting in 91 can be simplified greatly.